extensions/sql-migration/src/constants/helper.ts

```diff
@@ -158,41 +158,48 @@ export function hasMigrationOperationId(migration: DatabaseMigration | undefined
 export function canCancelMigration(migration: DatabaseMigration | undefined): boolean {
 	const status = getMigrationStatus(migration);
 	return hasMigrationOperationId(migration)
-		&& (status === loc.MigrationStatus.InProgress ||
-			status === loc.MigrationStatus.Retriable ||
-			status === loc.MigrationStatus.Creating);
+		&& (status === loc.MigrationState.InProgress
+			|| status === loc.MigrationState.Retriable
+			|| status === loc.MigrationState.Creating
+			|| status === loc.MigrationState.ReadyForCutover
+			|| status === loc.MigrationState.UploadingFullBackup
+			|| status === loc.MigrationState.UploadingLogBackup
+			|| status === loc.MigrationState.Restoring);
 }
 
 export function canDeleteMigration(migration: DatabaseMigration | undefined): boolean {
 	const status = getMigrationStatus(migration);
-	return status === loc.MigrationStatus.Canceled
-		|| status === loc.MigrationStatus.Failed
-		|| status === loc.MigrationStatus.Retriable
-		|| status === loc.MigrationStatus.Succeeded;
+	return status === loc.MigrationState.Canceled
+		|| status === loc.MigrationState.Failed
+		|| status === loc.MigrationState.Retriable
+		|| status === loc.MigrationState.Succeeded;
 }
 
 export function canRetryMigration(migration: DatabaseMigration | undefined): boolean {
 	const status = getMigrationStatus(migration);
-	return status === loc.MigrationStatus.Canceled
-		|| status === loc.MigrationStatus.Retriable
-		|| status === loc.MigrationStatus.Failed
-		|| status === loc.MigrationStatus.Succeeded;
+	return status === loc.MigrationState.Canceled
+		|| status === loc.MigrationState.Retriable
+		|| status === loc.MigrationState.Failed
+		|| status === loc.MigrationState.Succeeded;
 }
 
 export function canCutoverMigration(migration: DatabaseMigration | undefined): boolean {
 	const status = getMigrationStatus(migration);
 	return hasMigrationOperationId(migration)
-		&& status === loc.MigrationStatus.InProgress
 		&& isOnlineMigration(migration)
-		&& isFullBackupRestored(migration);
+		&& (status === loc.MigrationState.ReadyForCutover || isFullBackupRestored(migration));
```

Comment:
After talking to LRS team, one feedback they gave was to disable cutover option if restore is blocked, since cutover would fail in that case. On the RP side, the change will be to include that check as part of determining ReadyForCutover, but looks like that will be needed here as well.

Reply:
got it, thanks for the heads up. Let's revisit this once we finalize how to coordinate the RP/UI release

Reply:
Updated